src/GafferArnoldUI/GoboVisualiser.cpp

```diff
@@ -0,0 +1,339 @@
+		{
+			Color3f c = colors->readable()[u];
+
+			r.push_back( c[0] );
+			g.push_back( c[1] );
+			b.push_back( c[2] );
+		}
+
+		Imath::Box2i dataWindow( Imath::V2i( 0.0f ), Imath::V2i( resolution - 1 ) );
+		Imath::Box2i displayWindow( Imath::V2i( 0.0f ), Imath::V2i( resolution - 1 ) );
+
+		IECore::ImagePrimitivePtr image = new IECore::ImagePrimitive( dataWindow, displayWindow );
+		image->variables["R"] = IECore::PrimitiveVariable( PrimitiveVariable::Vertex, rData );
+		image->variables["G"] = IECore::PrimitiveVariable( PrimitiveVariable::Vertex, gData );
+		image->variables["B"] = IECore::PrimitiveVariable( PrimitiveVariable::Vertex, bData );
+
+		return image;
+	}
+
+	return NULL;
+}
+
+} // namespace
+
+int GoboVisualiser::m_texCount = 0;
+
+GoboVisualiser::GoboVisualiser()
+{
+}
+
+GoboVisualiser::~GoboVisualiser()
+{
+}
+
+IECoreGL::ConstRenderablePtr GoboVisualiser::visualise( const IECore::InternedString &attributeName, const IECore::ObjectVector *shaderVector, const IECore::Shader *lightShader, IECoreGL::ConstStatePtr &state ) const
+{
+	IECoreGL::ConstRenderablePtr standardResult = StandardLightFilterVisualiser::visualise( attributeName, shaderVector, lightShader, state );
+
+	if( !shaderVector || shaderVector->members().empty() || !lightShader )
+	{
+		return standardResult;
+	}
+
+	const IECore::Shader *filterShader = IECore::runTimeCast<const IECore::Shader>( shaderVector->members().back().get() );
+	if( !filterShader )
+	{
+		return standardResult;
+	}
+
+	CompoundDataMap::const_iterator it = filterShader->parameters().find( "slidemap" );
+	if( it == filterShader->parameters().end() )
+	{
+		return standardResult;
+	}
+
+	// Generate unique texture.
+	// TODO:   This will flood the texture cache and it shouldn't be used like this at all.
+	//         This is doing the trick for a quick prototype, though.
+	m_texCount += 1;
+	std::string textureName = boost::str( boost::format( "%s.oslmem" ) % m_texCount );
+
+	int resolution = 512;
+
+	const StringData *slidemap = IECore::runTimeCast<const StringData>( it->second.get() );
+	if( !slidemap )
+	{
+		const Color3fData *goboColor = IECore::runTimeCast<const Color3fData>( it->second.get() );
+		if( !goboColor )
+		{
+			return standardResult;
+		}
+
+		Imath::Box2i dataWindow( Imath::V2i( 0.0f ), Imath::V2i( resolution - 1 ) );
+		Imath::Box2i displayWindow( Imath::V2i( 0.0f ), Imath::V2i( resolution - 1 ) );
+
+		GafferOSL::OSLTextureReader::storeTexture( textureName, ImagePrimitive::createRGB( goboColor->readable(), dataWindow, displayWindow ) );
+	}
+	else
+	{
+		// \todo Some of this should be moved into the evaluator
+
+		// replace the root of the network with a surface shader that is correctly connected.
+		IECore::Shader *surfaceShader = new IECore::Shader( "Surface/Constant", "osl:shader" );
+		surfaceShader->parameters()["Cs"] = new StringData( slidemap->readable() );
+
+		IECore::ObjectVector *oslNetwork = new IECore::ObjectVector();
```

Comment:
Memory leak? Always assign the result of `new` to a smart pointer of some sort - in this case ObjectVectorPtr. As of C++14 they're pretty much saying that you should never call `new`, because you can use `make_unique` and `make_shared` instead. I'm thinking about doing something to enforce the creation of RefCounted classes in a similar way, but haven't got there yet...